contrib/epee/include/misc_language.h

```diff
@@ -128,6 +128,59 @@ namespace misc_utils
 
   }
 
+  template<typename T>
+  T factorial(T n, T s = 0)
```

Comment:
Why make this templated? The difficult part is that since its not used anywhere (right now), I have no idea whether the template is useful or not. It _seems like_ only one type will eventually be necessary.

Each type ends up being its own assembly, so its best to converge where possible.

Reply:
It feels unpleasant to tailor an otherwise generic math function to one use-case.

Reply:
Why is it unpleasant? We probably only need to use unsigned integers of a certain width anyway.